src/GeoFabrics/geometry.py

```diff
@@ -0,0 +1,316 @@
+        
+        if self._raster_origin is None:
+            self._raster_origin = [self.catchment.loc[0].geometry.bounds[0], 
+                                   self.catchment.loc[0].geometry.bounds[1]]
+        return self._raster_origin
+
+    @property
+    def raster_size(self):
+        """ Return the size of the catchment bbox in raster cells given the 
+        resolution """
+        
+        if self._raster_size is None:
+            self._raster_size = [int((self.catchment.loc[0].geometry.bounds[2] - 
+                                     self.catchment.loc[0].geometry.bounds[0]) / self.resolution), 
+                                int((self.catchment.loc[0].geometry.bounds[3] - 
+                                     self.catchment.loc[0].geometry.bounds[1]) / self.resolution)]
+        return self._raster_size
+    
+    @property
+    def catchment(self):
+        """ Return the catchment region """
+        
+        return self._catchment
+    
+    @property
+    def land(self):
+        """ Return the catchment land region """
+        
+        return self._land
+    
+    @property
+    def foreshore(self):
+        """ Return the catchment foreshore region """
+        
+        if self._foreshore is None:
+            self._foreshore = geopandas.overlay(self.land_and_foreshore, 
+                                                self.land, how='difference')
+        return self._foreshore
+    
+    @property    
+    def land_and_foreshore(self):
+        """ Return the catchment and and foreshore region """
+        
+        if self._land_and_foreshore is None:
+            self._land_and_foreshore = geopandas.GeoDataFrame(index=[0], geometry=self.land.buffer(self.resolution * self.foreshore_buffer), crs=self.crs)
+            self._land_and_foreshore = geopandas.clip(self.catchment, self._land_and_foreshore)
+        return self._land_and_foreshore
+    
+    @property
+    def foreshore_and_offshore(self):
+        """ Return the catchment land and offshore region """
+        
+        if self._foreshore_and_offshore is None:
+            self._foreshore_and_offshore = geopandas.overlay(self.catchment, 
+                                                self.land, how='difference')
+        return self._foreshore_and_offshore
+    
+    @property
+    def offshore(self):
+        """ Return the catchment offshore region """
+        
+        if self._offshore is None:
+            self._offshore = geopandas.overlay(self.catchment, 
+                                                self.land_and_foreshore, how='difference')
+        return self._offshore
+    
+    @property
+    def lidar_extents(self):
+        """ Return the catchment lidar extents """
+        
+        assert self._lidar_extents is not None, "lidar_extents have not been set, and need to be set explicitly"
+            
+        return self._lidar_extents
+    
+    def load_lidar_extents(self, lidar_extents_string): # expect to be filepath
```

Comment:
You could do this type comment as a type hint such as in [type hint](https://docs.python.org/3/library/typing.html)  or other ways with in-built Python support

Reply:
Good idea. I've added for most of the class constructors as well